go/common/keyformat/hashed.go

```diff
@@ -0,0 +1,80 @@
+package keyformat
+
+import (
+	"encoding"
+	"fmt"
+
+	"github.com/oasislabs/oasis-core/go/common/crypto/hash"
+)
+
+// PreHashed represents a pre-hashed value that will be encoded without additional hashing when used
+// together with a hashed format.
+type PreHashed hash.Hash
+
+// MarshalBinary encodes a hash into binary form.
+func (ph *PreHashed) MarshalBinary() ([]byte, error) {
+	return (*hash.Hash)(ph).MarshalBinary()
+}
+
+// UnmarshalBinary decodes a binary marshaled hash.
+func (ph *PreHashed) UnmarshalBinary(data []byte) error {
+	return (*hash.Hash)(ph).UnmarshalBinary(data)
+}
+
+// Equal compares vs another hash for equality.
+func (ph *PreHashed) Equal(cmp *PreHashed) bool {
+	return (*hash.Hash)(ph).Equal((*hash.Hash)(cmp))
+}
+
+// String returns the string representation of a hash.
+func (ph PreHashed) String() string {
+	return hash.Hash(ph).String()
+}
+
+type hashedFormat struct {
+	inner interface{}
+}
+
+func (h *hashedFormat) getData(v interface{}) ([]byte, error) {
+	switch t := v.(type) {
+	case encoding.BinaryMarshaler:
+		return t.MarshalBinary()
+	case []byte:
+		return t, nil
+	default:
+		panic(fmt.Sprintf("unsupported type: %T", t))
```

Comment:
If this can return an error, and the only caller returns an error, should this really be panicing?